modules/kubernetes-addons/karpenter/data.tf

```
@@ -48,7 +52,7 @@ data "aws_iam_policy_document" "karpenter" {
         "sqs:GetQueueUrl",
         "sqs:ReceiveMessage",
       ]
-      resources = [var.sqs_queue_arn]
+      resources = [data.aws_sqs_queue.karpenter.arn]
```

which means this will become 
```suggestion
      resources = [data.aws_sqs_queue[0].this.arn]
```